### PR TITLE
Part 1 of the multi-select chip

### PR DESCRIPTION
**Pull request by rupalg**
This widget allows the user to select a given number of options by tapping on the chips.

**Part 1 of the widget implements the following functionality**
- The two main parts of this widget are: text at the top of the list, and the list itself.
- The generic widget currently allows passing parameters for the title, and the list of chips (each chip has an optional icon and a required text).
Screenshot 2023-09-12 at 10 37 37 AM
Screenshot 2023-09-12 at 10 37 27 AM

**Part 2 of this widget would iterate on this chip to do the following:**
- the generic widget would allow passing the parameter for the maximum number of selections
- If the maximum number of chips is selected, tapping on additional chips has no effect. The user can always deselect a previous selection and then select a new one.

### CODE REVIEW

File: lib/widgets/molecules/multi_select_chips.dart [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    required this.chips,
+    this.maxSelection,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    String textString = maxSelection != null
+        ? "Select up to $maxSelection (optional)"
+        : "Select all that apply (optional)";
```
Comment (m-rosario) [inline lib/widgets/molecules/multi_select_chips.dart:65]: Define these as AppLocalizations in l10n.

File: lib/widgets/molecules/multi_select_chips.dart [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          child: Text(
+            textString,
+            style: theme.textTheme.bodyMedium?.copyWith(
+              color: theme.colorScheme.secondary,
+            ),
+          ),
+        ),
+        const SizedBox(
+          height: 20,
+        ),
```
Comment (m-rosario) [inline lib/widgets/molecules/multi_select_chips.dart:79]: This is essentially being used as padding, so you could use one of the predefined Insets.padding constants.

**Review by m-rosario — CHANGES_REQUESTED** [review]
This look great, but unfortunately you implemented the old version of the design. 😓 The new versions are at the bottom in Figma, here is a picture:

image

I am sorry if this design revision wasn't made clear in the task. Thankfully there are not too many changes that need to be done.

**Comment by rupalg** [comment]
I thought that the trello screenshots were the old version: https://trello.com/c/Tx7XW3gI/108-p0-signup-ui-components

Well, thanks for the explicit crosses above. I will work on the redesign.

**Comment by rupalg** [comment]
Hey there, 
I have made the needed UI changes:
Screenshot 2023-09-12 at 11 42 20 AM

Screenshot 2023-09-12 at 11 42 07 AM

**Review by m-rosario — APPROVED** [review]
Excellent!!!

**Pull request merged**